src/Admin/SupportScreen.php

```diff
@@ -170,8 +173,8 @@ public function enqueue_assets( $hook_suffix ) {
 			];
 		}
 
-		$this->support_data->set_args( $args );
-		$data = $this->support_data->get_data();
+		$support_data = $this->injector->make( SupportData::class, [ 'args' => $args ] );
```

Comment:
After taking a closer look, I think instantiating `SupportData` via the injector makes things overly complicated. Directly creating a `SupportData` object should be simple enough here, and therefore there would be no need for `Injector`.

```suggestion
		$support_data = new SupportData( $args );
```

Reply:
Another alternative:

```suggestion
		$support_data = $this->injector->make( SupportData::class, compact( 'args' ) );
```